tests/unit/ceed/test_ceed.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "catch.hpp"
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+#include "../../../fem/libceed/ceed.hpp"
+
+using namespace mfem;
+
+namespace ceed_test
+{
+
+double coeff_function(const Vector &x)
+{
+   return 1.0 + x[0]*x[0];
+}
+
+static std::string getString(AssemblyLevel assembly)
+{
+   switch (assembly)
+   {
+   case AssemblyLevel::NONE:
+      return "NONE";
+      break;
+   case AssemblyLevel::PARTIAL:
+      return "PARTIAL";
+      break;
+   case AssemblyLevel::ELEMENT:
+      return "ELEMENT";
+      break;
+   case AssemblyLevel::FULL:
+      return "FULL";
+      break;
+   case AssemblyLevel::LEGACYFULL:
+      return "LEGACYFULL";
+      break;
+   }
+}
+
+static std::string getString(CeedCoeff coeff_type)
+{
+   switch (coeff_type)
+   {
+   case CeedCoeff::Const:
+      return "Const";
+      break;
+   case CeedCoeff::Grid:
+      return "Grid";
+      break;
+   case CeedCoeff::Quad:
+      return "Quad";
+      break;
+   }
+}
+
+enum class Problem {Mass, Diffusion, VectorMass, VectorDiffusion};
+
+static std::string getString(Problem pb)
+{
+   switch (pb)
+   {
+   case Problem::Mass:
+      return "Mass";
+      break;
+   case Problem::Diffusion:
+      return "Diffusion";
+      break;
+   case Problem::VectorMass:
+      return "VectorMass";
+      break;
+   case Problem::VectorDiffusion:
+      return "VectorDiffusion";
+      break;
+   }
+}
```

Comment:
`warning: control reaches end of non-void function`

Reply:
done